### ISSUE TITLE
Release version with an unescaped string.

### ISSUE THREAD

**Issue by deissh** [getsentry/sentry#25538]
## Important Details

How are you running Sentry?

Saas (sentry.io)

## Description

error when trying to open a release. also, when trying to delete via the API, it gives the answer that it was not found
![image](https://user-images.githubusercontent.com/21129524/115654541-03ac8280-a33a-11eb-8fc0-0343479390fa.png)

## Steps to Reproduce

1. create a new release
2. then you need to set the release version, for example, I have problems with this name `"\n> rf-frontend@0.1.0 release:version\n> echo "dev-19be1b7e-dirty"\n\ndev-19be1b7e-dirty"` (forgot to escape the symbol in the bash script)

Good items to include here include:

```
vendor.f93d44c7de6da4a40abf.js:2 GET https://sentry.io/api/0/organizations/satek-o5/sessions/?project=5256760&query=release%3A%22%5Cn%3E+rf-frontend%400.1.0+release%3Aversion%5Cn%3E+echo+%22dev-19be1b7e-dirty%22%5Cn%5Cndev-19be1b7e-dirty%22&field=sum(session)&statsPeriod=90d&interval=1d 400 (Bad Request)

{"detail":"Invalid query field: \"message\""}
```

**Comment by untitaker** [comment]
how are you creating the release?

**Comment by deissh** [comment]
> how are you creating the release?

sentry-cli

**Comment by untitaker** [comment]
Is this blocking you in any way when using the product or are you just trying to fuzz something? I have problems understanding why you'd want to create a release with a newline or quotation marks in it.

**Comment by BYK** [comment]
@untitaker I think they created this by accident:

>(forgot to escape the symbol in the bash script)

We should still reject these kinds of values?

**Comment by untitaker** [comment]
Well I also don't understand why one would create a release with a properly escaped `\n`.

EDIT: Ah ok, seems like it was not intended to create any of that.

We'll have to pull out or port the logic from Relay to make sure we're using the same thing everywhere. You won't be able to create that kind of release via sending an event.

**Comment by matejminar** [comment]
@kamilogorek should we open an issue against https://github.com/getsentry/sentry-cli repo?

**Comment by ahmedetefy** [comment]
Here are my findings on this so far:-
This is an issue on both the SDK side and the API side

**So for the SDK side,**
The issue here is that you can (atleast for Python and JS, these are the ones that I have tested) send sessions with incorrect release versions (with bad chars in the string) like the one above, and they will be stored in Snuba 

In JS for example, we are only checking if release is a string but we should probably check for bad chars as well similar to what we do here on sentry -> https://github.com/getsentry/sentry/pull/26279

Proposed solution:- 
- Validate release name on `Sentry.init` in SDKs, or atleast drop sessions if release string contains bad characters or so 

**For the API side (which is basically called by the cli),** 
Direct creation of release (POST) has correct validation applied on it so this issue is not caused by that 

However, in the case if you send healthy sessions for example associated with a new release version, the release is not created right away but rather only created the first time you go to the releases list page and there is no validation is applied there.

So what I am assuming happened is some healthy sessions for example were sent associated with that non escaped incorrect release version string -> an API call was made to the releases list page, and since this new release has associated health data in Snuba (sessions data) -> that created a new release for it, and because there is no validation there, the release creation was not rejected.

Proposed Solution:
- I will add validation so that if releases list page is called, and there is some health data associated with a release version that has not been created yet -> Creation will be rejected if release string is not valid

TL;DR
I spoke with Armin and he suggested that we should not change it on sentry or the sdks but rather on relay

**Comment by jan-auer** [comment]
Agreed that this needs to be fixed in Relay, transferring.
